src/main/java/co/rsk/federate/signing/hsm/client/HSMClientProtocol.java

```diff
@@ -100,35 +99,35 @@ public ObjectNode buildCommand(String commandName, int version) {
     public JsonNode send(ObjectNode command) throws HSMClientException {
         int attempts = 0;
         JsonRpcClient client = null;
-        while(true) {
+        while (true) {
             try {
                 client = clientProvider.acquire();
                 String commandName = command.get("command").toString();
-                logger.trace("Sending command to hsm: {}", commandName);
+                logger.trace("[send] Sending command to hsm: {}", commandName);
                 Future future = getExecutor().submit(new HSMRequest(client, command));
                 JsonNode result = null;
                 try {
-                    logger.trace("Fetching response for command: {}", commandName);
+                    logger.trace("[send] Fetching response for command: {}", commandName);
                     result = (JsonNode) future.get();
-                    logger.trace("Got response for command: {}", commandName);
+                    logger.trace("[send] Got response for command: {}", commandName);
                 } catch (ExecutionException e) {
                     Throwable cause = e.getCause();
                     if (cause instanceof JsonRpcException) {
-                        throw (JsonRpcException)cause;
+                        throw (JsonRpcException) cause;
                     }
                     if (cause instanceof HSMDeviceNotReadyException) {
-                        throw (HSMDeviceNotReadyException)cause;
+                        throw (HSMDeviceNotReadyException) cause;
                     }
                     if (cause instanceof HSMClientException) {
-                        throw (HSMClientException)cause;
+                        throw (HSMClientException) cause;
                     }
```

Comment:
What if the cause isn't any of those exception types? `result` method variable will be null, and the code will continue until the end. Why not add a default when the cause is none of those types?

Reply:
This Is Out Of The Scope For This Task, However, There's An External Try Catch To Handle It

Reply:
The external try catch will handle it **only if** the exception is not an `ExecutionException,` otherwise the flow will continue and call `validateResponse` passing a null.

Reply:
I am closing this discussion since we agreed to handle this in another ticket.